### PR TITLE
Add support for PHP 8.3

### PR DESCRIPTION
**Pull request by gsteel**
Also adds recommended settings for psalm and deals with a few minor issues

### CODE REVIEW

File: src/AutoComposer.php [hunk 1]
```diff
@@ -34,12 +34,6 @@ class AutoComposer
     /** @var resource */
     private $errorStream;
 
-    /** @var string[] */
-    private array $expectedValues = [
-        '0', // production mode
-        '1', // development mode
-    ];
-
```
Comment (gsteel) [inline src/AutoComposer.php:42]: Completely unused private property

File: test/CommandTest.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\DevelopmentMode;
+
+use Laminas\DevelopmentMode\Command;
+use PHPUnit\Framework\TestCase;
+
+class CommandTest extends TestCase
```
Comment (Ocramius) [inline test/CommandTest.php]: `final` + `@covers` :D

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/CommandTest.php (hunk 2 above)
Comment (gsteel) [reply test/CommandTest.php]: Thanks @Ocramius

**Review by Ocramius — APPROVED** [review]
Thanks @gsteel!

**Pull request merged**